### PR TITLE
[SPARK-29403][INFRA][R] Uses Arrow R 0.14.1 in AppVeyor for now

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to use Arrow R 0.14.1 for now in AppVeyor to make tests passed.

### Why are the changes needed?

To make build passed with Arrow. It doesn't work with setting `ARROW_PRE_0_15_IPC_FORMAT` to `1` to allow Arrow R 0.15 compatibility.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

AppVeyor

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @BryanCutler FYI

**Comment by HyukjinKwon** [comment]
@BryanCutler did Arrow change the format itself? bunch of PySpark tests with Arrow fail too.

**Comment by HyukjinKwon** [comment]
Seems Arrow isn't released yet in Windows. The AppVeyor build here used Areow 0.14.1.1. I'll trigger the build again when Areow 0.15 is available on Windows.

**Comment by harpaj** [comment]
Might have the same cause as Pandas UDFs not working with Arrow 0.15, see https://issues.apache.org/jira/projects/SPARK/issues/SPARK-29367

**Comment by BryanCutler** [comment]
> @BryanCutler did Arrow change the format itself? bunch of PySpark tests with Arrow fail too.

Yes, there was a change but I have a patch ready that will be backwards compatible. I'll open a JIRA for the upgrade and we can go over the details.

**Comment by BryanCutler** [comment]
I opened a PR documenting the change in #26045

**Comment by HyukjinKwon** [comment]
Hm, setting `ARROW_PRE_0_15_IPC_FORMAT` to `1` doesn't seem working in SparkR case. I tested this both in Windows (via AppVeyor) and Ubuntu

**Comment by HyukjinKwon** [comment]
I am going to set the Arrow R version to 0.14.1 for now explicitly to avoid to break build. The build is borken in the master due to this.

File: appveyor.yml [hunk 1]
```diff
@@ -57,6 +57,8 @@ build_script:
 
 environment:
   NOT_CRAN: true
+  # See SPARK-29378.
+  ARROW_PRE_0_15_IPC_FORMAT: 1
```
Comment (BryanCutler) [inline appveyor.yml]: Does setting this here carry it through to the worker?

Comment (HyukjinKwon) [reply appveyor.yml]: I removed it in the latest commit :-).

**Comment by HyukjinKwon** [comment]
I am merging this to keep our AppVeyor build alive for now ..

**Comment by dongjoon-hyun** [comment]
Hi, All.
After we upgrade Arrow 0.15.1, AppVeyor is dead again~ I'm revisiting this issue in the following PR.
- https://github.com/apache/spark/pull/26555

**Pull request closed without merging**